### PR TITLE
Remove `isEncryptedValue` from `WebXSecret`s

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/sms/secret/web3/Web3Secret.java [hunk 1]
```diff
@@ -31,34 +31,23 @@ public class Web3Secret extends Secret {
...
-     * Copies the current {@link Web3Secret} object,
-     * while replacing the old value with a new decrypted value.
-     *
-     * @param newDecryptedValue Value to use for new object.
-     * @return A new {@link Web3Secret} object with new value.
-     */
-    public Web3Secret withDecryptedValue(String newDecryptedValue) {
-        return new Web3Secret(header, newDecryptedValue, false);
+    public Web3Secret withValue(String newValue) {
+        return new Web3Secret(header, newValue);
```
[the remark this answers is not in the document]
Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/web3/Web3Secret.java]: Good point! Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/146/commits/287ada21bd99bb94cbd34276d9c0154137fd85a1.

File: src/test/java/com/iexec/sms/secret/web2/Web2SecretTests.java [hunk 2]
```diff
@@ -27,25 +27,25 @@ class Web2SecretTests {
 
     private static final String UNENCRYPTED_VALUE = "unencryptedValue";
     private static final String ENCRYPTED_VALUE   = "encryptedValue";
-    private static final Web2Secret UNENCRYPTED_SECRET = new Web2Secret(OWNER_ADDRESS, SECRET_ADDRESS, UNENCRYPTED_VALUE, false);
-    private static final Web2Secret ENCRYPTED_SECRET   = new Web2Secret(OWNER_ADDRESS, SECRET_ADDRESS, ENCRYPTED_VALUE, true);
+    private static final Web2Secret UNENCRYPTED_SECRET = new Web2Secret(OWNER_ADDRESS, SECRET_ADDRESS, UNENCRYPTED_VALUE);
+    private static final Web2Secret ENCRYPTED_SECRET   = new Web2Secret(OWNER_ADDRESS, SECRET_ADDRESS, ENCRYPTED_VALUE);
 
     @Test
     void withEncryptedValue() {
```
[the remark this answers is not in the document]
Comment (mcornaton) [reply src/test/java/com/iexec/sms/secret/web2/Web2SecretTests.java]: Right. Here it is: https://github.com/iExecBlockchainComputing/iexec-sms/pull/146/commits/033113dab23ff039bbf8c037205b896ca04d4f25.

File: src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java [hunk 3]
```diff
@@ -37,20 +37,33 @@ protected Web2SecretService(EncryptionService encryptionService,
...
+     * @param secretAddress Address of the secret.
+     * @return An empty {@link Optional} if no secret is found,
+     * an {@link Optional} containing the secret if it exists.
+     */
+    Optional<Web2Secret> getSecret(String ownerAddress, String secretAddress) {
         return web2SecretRepository.findById(new Web2SecretHeader(ownerAddress, secretAddress));
     }
 
-    public Optional<Web2Secret> getSecret(String ownerAddress, String secretAddress, boolean shouldDecryptValue) {
+    public Optional<String> getDecryptedValue(String ownerAddress, String secretAddress) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java:53]: We might simplify them here or later:
```
    public Optional<String> getDecryptedValue(String ownerAddress, String secretAddress) {
        return getSecret(ownerAddress, secretAddress)
            .map(secret -> encryptionService.decrypt(secret.getValue()));
    }
```

File: src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java [hunk 4]
```diff
@@ -65,14 +78,14 @@ public Optional<Web2Secret> getSecret(String ownerAddress, String secretAddress,
      *                                      with same {@code ownerAddress}/{@code secretAddress} couple already exists
      */
     public Web2Secret addSecret(String ownerAddress, String secretAddress, String secretValue) throws SecretAlreadyExistsException {
-        final Optional<Web2Secret> oSecret = getSecret(ownerAddress, secretAddress);
-        if (oSecret.isPresent()) {
+        final boolean isSecretPresent = isSecretPresent(ownerAddress, secretAddress);
+        if (isSecretPresent) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java:82]:
```suggestion
        if (isSecretPresent(ownerAddress, secretAddress)) {
```
Same applies for other parts

Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java]: Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/146/commits/5759ec2812187cbfbd5be50ab5a874b01802e210.

**Review by jbern0rd — APPROVED** [review]
Nice work

**Review by jeremyjams — APPROVED** [review]
Thank you for this work!

**Pull request merged**